src/plugins/intel_gpu/src/graph/compilation_context.cpp

```diff
@@ -1,91 +1,74 @@
         std::lock_guard<std::mutex> lock(_mutex);
-        if (_queue_keymap.find(removed_key) != _queue_keymap.end()) {
-            _queue_keymap.erase(removed_key);
+        if (!_task_keys.empty()) {
+            for (auto key : keys) {
+                if (_task_keys.find(key) != _task_keys.end()) {
+                    _task_keys.erase(key);
+                }
+            }
         }
     }
 
-private:
-    std::deque<CompilationTaskData> _queue;
-    std::unordered_map<size_t, std::deque<CompilationTaskData>::iterator> _queue_keymap;
-    std::mutex _mutex;
-};
-
-class CompilationContext : public ICompilationContext {
-public:
-    CompilationContext(cldnn::engine& engine, const ExecutionConfig& config, size_t program_id) {
-        _kernels_cache = cldnn::make_unique<kernels_cache>(engine, config, program_id, nullptr, kernel_selector::KernelBase::get_db().get_batch_header_str());
-        _worker = std::thread([this](){
-            while (!_stop_compilation) {
-                CompilationContext::Task task;
-                size_t task_key;
-                bool success = _queue.pop_front_task(task_key, task);
-                if (success) {
-                    task(*_kernels_cache);
-                    _queue.erase_task_key(task_key);
-                } else {
-                    std::chrono::milliseconds ms{1};
-                    std::this_thread::sleep_for(ms);
-                }
-            }
-        });
+    ~CompilationContext() noexcept {
+        cancel();
     }
 
-    void push_task(size_t key, ICompilationContext::Task&& task) override {
-        _queue.push_task(key, std::move(task));
+    bool is_stopped() override {
+        return _stop_compilation;
     }
 
     void cancel() noexcept override {
+        if (_stop_compilation)
+            return;
+
         _stop_compilation = true;
-        if (_worker.joinable())
-            _worker.join();
+        if (_task_executor != nullptr) {
+            _task_executor.reset();
+            _task_executor = nullptr;
+        }
+        _task_keys.clear();
     }
 
-    ~CompilationContext() noexcept { cancel(); }
-
 private:
-    std::unique_ptr<kernels_cache> _kernels_cache;
-    std::thread _worker;
+    InferenceEngine::CPUStreamsExecutor::Config _task_executor_config;
+    InferenceEngine::CPUStreamsExecutor::Ptr _task_executor;
+    std::mutex _mutex;
+    std::unordered_set<size_t> _task_keys;
     std::atomic_bool _stop_compilation{false};
-
-    CompilationTaskQueue _queue;
 };
 
-std::unique_ptr<ICompilationContext> ICompilationContext::create(cldnn::engine& engine, const ExecutionConfig& config, size_t program_id) {
-    return cldnn::make_unique<CompilationContext>(engine, config, program_id);
+std::unique_ptr<ICompilationContext> ICompilationContext::create(InferenceEngine::CPUStreamsExecutor::Config task_executor_config) {
+    return cldnn::make_unique<CompilationContext>(task_executor_config);
 }
 
-}  // namespace cldnn
+}  // namespace cldnn
```

Comment:
Please add a blank line at the end of file.

Reply:
Done